pkg/blobinfocache/memory/memory.go

```diff
@@ -160,25 +171,45 @@ func (mem *cache) CandidateLocations(transport types.ImageTransport, scope types
 // data from previous RecordDigestUncompressedPair calls is used to also look up variants of the blob which have the same
 // uncompressed digest.
 func (mem *cache) CandidateLocations2(transport types.ImageTransport, scope types.BICTransportScope, primaryDigest digest.Digest, canSubstitute bool) []blobinfocache.BICReplacementCandidate2 {
-	return mem.candidateLocations(transport, scope, primaryDigest, canSubstitute, true)
+	resAllBlobs := []prioritize.CandidateWithTime{}
```

Comment:
BTW all of this would need to be done under the existing lock. Another reason to do that _inside_ `candidateLocations` instead of duplicating it.

Reply:
I re-acquired lock in CandidateLocations2 just after calling `candidateLocations`, i think since allblobs is needed and this is only memory so I can't think of any additional harmful race between two invocations.

Reply:
This is merged into `candidateLocations` so this comment is not relevent anymore.